ledger/src/main/java/com/scalar/dl/ledger/database/scalardb/ScalarNamespaceResolver.java

```diff
@@ -0,0 +1,45 @@
+package com.scalar.dl.ledger.database.scalardb;
+
+import com.google.inject.Inject;
+import com.scalar.dl.ledger.config.ServerConfig;
+import com.scalar.dl.ledger.namespace.NamespaceManager;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A resolver that converts logical namespace names to physical ScalarDB namespace names.
+ *
+ * <p>The conversion rule is:
+ *
+ * <ul>
+ *   <li>Default namespace -&gt; base namespace from ServerConfig
+ *   <li>Other namespaces -&gt; base namespace + "_" + logical namespace
```

Comment:
Just curious, what's the reason for this special treat for the default namespace, instead of returning "${baseNamespace}_${defaultNamespace}" as other namespaces?